### PR TITLE
Js webapp details

### PR DESCRIPTION
**Pull request by jspaaks**
This PR is another small iteration.
Biggest change is that I removed the index.html snippet near the top (the one on the bottom is still there), added an image of Newton, and added a link to an explanation of Newton-Raphson on YouTube.

Also I changed the contents of index.html a bit: removed things that were not immediately relevant to the story, limited the ``head`` to only the ``newtonraphson.js`` ``src``, the other script tag is now in the ``body``. In that script, I also changed the name ``module`` to ``NewtonRaphson``, to mimic the constructor syntax used in C++. Finally, I also changed from using a ``span`` to a ``div``, as that is more in line with how React and Vue etc work.

### CODE REVIEW

File: js-webapp/README.md [hunk 1]
```diff
@@ -9,40 +9,19 @@ _Wouldn't it be great if you could run your existing C/C++ code on the web with
...
 application. Subsequent blogs in this series will expand on the current one by laying out more advanced topics,
 specifically how to [make the app interactive](anotherblog), how to [visualize the results](someotherblog), and
 how to [deal with long running tasks](yetanotherblog).
 
-So today's aim is to have a simple web app like the code snippet below. At first glance this is just some boilerplate to
-be able to use the ``newtonraphson.js`` library, but **the neat part is that the complete Newton-Raphson code is written
-in C++, not JavaScript**. With some trickery, we'll be able to use that C++ code from the browser, without the need to
-port it first!
+So today's aim is to have a simple web app that determines the _root_ of a mathematical function _x^3 - x^2 + 2_, i.e. the
+value of _x_ where _y = 0_. For this, we'll use the [_Newton-Raphson_ root finding method](https://www.youtube.com/watch?v=cOmAk82cr9M). Remember Newton? Quiet
```
Comment (sverhoeven) [inline js-webapp/README.md]: You now explain the equation, a picture like https://nlesc-jcer.github.io/cpp2wasm/images/equation.svg with a big arrow to -1,0 coordinate would help me here.

File: js-webapp/README.md [hunk 2]
```diff
@@ -9,40 +9,19 @@ _Wouldn't it be great if you could run your existing C/C++ code on the web with
...
-                            root.toFixed(2);
-         });
-      </script>
-   </head>
-   <body>
-      <span id="answer"> </span>
-   </body>
-</html>
-```
+![newton.jpg](newton.jpg)
```
Comment (sverhoeven) [inline js-webapp/README.md:33]: Don't forget to add attribution info for images.

File: js-webapp/README.md [hunk 3]
```diff
@@ -187,30 +168,27 @@ this JavaScript library, we can find the root of the mathematical function, and
...
    <head>
-      <title>Newton-Raphson</title>
-      <meta charset="utf-8">
       <script type="text/javascript" src="newtonraphson.js"></script>
+   </head>
+   <body>
+      <div id="answer"></div>
       <script>
-         createModule().then((module) => {
+         createModule().then((NewtonRaphson) => {
```
Comment (sverhoeven) [inline js-webapp/README.md]: I would use `rootfinding` instead as that is the C++ namespace in which the class is defined.
```suggestion
         createModule().then((rootfinding) => {
```

File: js-webapp/README.md [hunk 4]
```diff
@@ -187,30 +168,27 @@ this JavaScript library, we can find the root of the mathematical function, and
...
             const tolerance = 0.001;
-            const newtonraphson = new module.NewtonRaphson(tolerance);
             const initial_guess = -20;
+            const newtonraphson = new NewtonRaphson.NewtonRaphson(tolerance);
```
Comment (sverhoeven) [inline js-webapp/README.md]:
```suggestion
            const newtonraphson = new rootfinding.NewtonRaphson(tolerance);
```

File: js-webapp/index.html [hunk 5]
```diff
@@ -1,22 +1,19 @@
...
    <head>
-      <title>Newton-Raphson</title>
-      <meta charset="utf-8">
       <script type="text/javascript" src="newtonraphson.js"></script>
+   </head>
+   <body>
+      <div id="answer"></div>
       <script>
-         createModule().then((module) => {
+         createModule().then((NewtonRaphson) => {
```
Comment (sverhoeven) [inline js-webapp/index.html]:
```suggestion
         createModule().then((rootfinding) => {
```

File: js-webapp/newton.jpg.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+https://commons.wikimedia.org/wiki/File:Sir_Isaac_Newton._Wellcome_V0006785EL.jpg
```
Comment (sverhoeven) [inline js-webapp/newton.jpg.txt:1]: What is the idea of this file? Will this be alt-text or caption of the image on medium?

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Less teasers, but that is indeed some nice hair.

File: js-webapp/README.md (hunk 2 above)
Comment (jspaaks) [reply js-webapp/README.md:33]: yep that's why I added the <filename>.txt files in the repo so I know where they come from.

File: js-webapp/README.md (hunk 3 above)
Comment (jspaaks) [reply js-webapp/README.md]: done

File: js-webapp/README.md (hunk 1 above)
Comment (jspaaks) [reply js-webapp/README.md]: good point. Maybe we should provide this feedback even before adding user interaction or long running tasks. Some simple d3 would do. in addition to adding the image near the top of the blog I mean.

**Review by sverhoeven — APPROVED** [review]
Good for now, will do plot image later, see #14

**Comment by jspaaks** [comment]
Note I just pushed updated content with an image

refs #14

**Pull request merged**